### DISCUSSION TITLE
How to configure shell like autosuggestions (shown in the demo)

### DISCUSSION THREAD

**Discussion by tmummert** [hrsh7th/nvim-cmp#1542] [Q&A] [answered]
Dear community,

in the demo video you can see shell like autosuggestions in the cursor line (i.e. second 3 `table`-field, dimmed characters to the right of the cursor). How can this be configured?

I have come across the function `cmp.complete_common_string()`, but this does not seem to be what i am looking for in this context.

Thank you for any help!

**Comment by Shougo** [comment]
It seems ghost text feature.

**Reply by tmummert** [reply]
I completely overlooked this feature, sorry!

Is it possible to complete the whole body of the entry (function, method) as shown in the documentation window, not just the abbreviation?

I.e. `insert(list, pos, value)` as `function table.insert(list: table, pos: integer, value: any)`?

**Reply by yutkat** [reply]
I don't use this feature but I tried it. It seems like this feature doesn't work now.	
Maybe it's buggy because I don't get any type information from `vim.lsp.util.parse_snippet`. 
https://github.com/neovim/neovim/blob/eb4676c67f5dd54bcda473783315901a3444b40b/runtime/lua/vim/lsp/util.lua#L615

I think we can do the same thing with this plugin, so you don't have to display it with ghost text.
https://github.com/hrsh7th/cmp-nvim-lsp-signature-help

**Reply by yutkat** [reply, marked as answer]
https://github.com/hrsh7th/nvim-cmp/discussions/544#discussioncomment-1662551

**Reply by tmummert** [reply]
this is exactly what i was looking for, thank you very much!

The setup must be done using the language server configuration.

You can find more information in the `lua-language-server`'s [documentation](https://github.com/LuaLS/lua-language-server/blob/d7e59826b5dec2d6f4b9ba59f6d64af42c078edd/locale/en-us/setting.lua#L143).

So in the [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig)-file add:

```
require'lspconfig'.lua_ls.setup {
  settings = {
    Lua = {
      ...
      completion = {
        callSnippet = "Replace",
      },
      ...
    },
  },
}
```
